src/components/BookDetails.js

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { useParams } from "react-router-dom";
+
+const BookDetails = ({ books }) => {
+  
+  const bookId = useParams().id;
```

Comment:
You can also destructure here, not sure if you'd want to but it's an option! `const {id} = userParams()`